families/track_and_trade/client/src/views/list_fish.js

```diff
@@ -0,0 +1,147 @@
+            'Updates'
+          ],
+          rows: vnode.state.filteredRecords.slice(
+            vnode.state.currentPage * PAGE_SIZE,
+            (vnode.state.currentPage + 1) * PAGE_SIZE)
+                .map((rec) => [
+                  truncate(rec.recordId, { length: 32 }),
+                  _getProp(rec, 'species'),
+                  // This is the "created" time, synthesized from properties
+                  // added on the initial create
+                  _formatTimestamp(rec.timestamp),
+                  _formatTimestamp(_getLatestPropUpdateTime(rec)),
+                  _countPropUpdates(rec)
+                ]),
+          noRowsText: 'No records found'
+        })
+      )
+    ]
+  }
+}
+
+const _controlButtons = (vnode, publicKey) => {
+  if (publicKey) {
+    let filterRecords = (f) => {
+      vnode.state.filteredRecords = vnode.state.records.filter(f)
+    }
+
+    return [
+      m('.col-sm-8',
+        m(FilterGroup, {
+          ariaLabel: 'Filter Based on Ownership',
+          filters: {
+            'All': () => { vnode.state.filteredRecords = vnode.state.records },
+            'Owned': () => filterRecords((record) => record.owner === publicKey),
+            'Custodian': () => filterRecords((record) => record.custodian === publicKey),
+            'Reporting': () => filterRecords(
+              (record) => record.properties.reduce(
+                (owned, prop) => owned || prop.reporters.indexOf(publicKey) > -1, false))
+          },
+          initialFilter: 'All'
+        })),
+      m('.col-sm-4', _pagingButtons(vnode))
+    ]
+  } else {
+    return [
+      m('.col-sm-4.ml-auto', _pagingButtons(vnode))
+    ]
+  }
+}
+
+const _pagingButtons = (vnode) =>
+  m(PagingButtons, {
+    setPage: (page) => { vnode.state.currentPage = page },
+    currentPage: vnode.state.currentPage,
+    maxPage: Math.floor(vnode.state.filteredRecords.length / PAGE_SIZE)
+  })
+
+const _getProp = (record, propName) => {
+  let prop = record.properties.find((prop) => prop.name === propName)
+  if (prop) {
+    return prop.value
+  } else {
+    return ''
+  }
+}
+
+const _getPropTimeByComparison = (compare) => (record) => {
+  if (!record.updates.properties) {
+    return null
+  }
+
+  return Object.entries(record.updates.properties)
+      .reduce((acc, [key, updates]) => acc.concat(updates), [])
+      .reduce((selectedTimestamp, update) =>
+              compare(selectedTimestamp, update.timestamp) ? update.timestamp : selectedTimestamp)
+}
+
+const _getLatestPropUpdateTime =
+  _getPropTimeByComparison((selected, timestamp) => selected < timestamp)
+
+const _countPropUpdates = (record) => {
+  if (!record.updates.properties) {
+    return 0
+  }
+
+  return Object.entries(record.updates.properties).reduce(
+    (sum, [key, updates]) => sum + updates.length, 0)
+}
+
+const _formatTimestamp = (sec) => {
+  if (!sec) {
+    sec = Date.now() / 1000
+  }
+  return moment(sec).format('MM/DD/YYYY, h:mm:ss a')
```

Comment:
<img width="576" alt="screen shot 2017-09-25 at 8 35 37 am" src="https://user-images.githubusercontent.com/8889580/30812981-ba22d906-a1d1-11e7-85f5-ea78dcad8398.png">

The wrong "Added" time seems to be cause by always being passed in as null, and then turned into seconds instead of milliseconds. I am not sure why the "Updated" time is getting the the day right, but erroneously reporting the time as midnight. Looking at the code it seems like you _should_ be correctly getting the latest update time (in seconds), and then turn it into 01/18/1970.